gk/fib.c

```diff
@@ -872,13 +965,35 @@ ether_cache_put(struct gk_fib *neigh_fib,
 	}
 
 	if (addr->proto == ETHER_TYPE_IPv4) {
-		return neigh_del_ether_cache(
+		ret = put_arp((struct in_addr *)
+			&addr->ip.v4, gk_conf->lcores[0]);
+		if (ret < 0)
+			return ret;
+
+		ret = neigh_del_ether_cache(
 			&neighbor_fib->u.neigh, &addr->ip.v4.s_addr);
+		if (ret < 0) {
+			RTE_LOG(WARNING, GATEKEEPER,
```

Comment:
We should use something with more priority than `WARNING` since we are not recovering from this failure.

It's appropriate to add a comment saying that we are not trying to recover.

ATTENTION: The same case for the IPv6 case below.